### DISCUSSION TITLE
Plugin Leveling System

### DISCUSSION THREAD

**Discussion by Ophise** [Pryaxis/TShock#2162] [Q&A]
Is there a plugin for a server level system?

**Comment by Kojirremer** [comment]
AutoRank did a pretty good job at it, but it's been abandoned for years, and since it depends on SEconomy anyways, there is an alternative. However, it is not a plugin per se, but rather a JIST script, JIST being a companion plugin of SEconomy.
I'm sorry to say, though, that I don't know where you can find the script right now. It may be surprisingly easy to find, or someone may provide it for you. Mine has been tweaked for spanish language quite a bit, so it won't be of use to you.

EDIT: I forgot to put the name of the script: "ranking-system-2.0.js"

**Reply by Ophise** [reply]
Is there an actual or at least a working version of SEconomy? I need any working ranked plugin for version 1.4.1.2

**Reply by Kojirremer** [reply]
https://github.com/Quinci135/SEconomy
Quinci's fork is working flawlessly for 1.4.1.2.

**Reply by Ophise** [reply]
Will AutoRank work with this plugin now?

**Reply by Ophise** [reply]
I am having trouble installing this plugin. What kind of files do you need to transfer to ServerPlugins?

**Reply by Kojirremer** [reply]
As I said, AutoRank is **DEAD**, unless someone updates it to both latest TShock and latest known working SEconomy build (which is currently Quinci's). The only **public** way to have a levelling system right now is the aforementioned JIST script.
As to how to configure SEconomy, it's not difficult, but since there are many things to set up it is still complicated. You need all DLL files here on your ServerPlugins directory:
https://github.com/Quinci135/SEconomy/tree/master/Builds

And if you want help with that... unless you ask about specific issues instead of "help configure everything", I'm afraid that's as far as I'll go helping you.

**Reply by Ophise** [reply]
Thanks for the answer, I figured it out, but I got another problem! I have an SSC system on my server. The plugin Seconomy can see my account perfectly, I can use its functionality, but ordinary players cannot use it. When you enter the command / bank bal, an answer is received that their account was not found and asks to log in (despite the fact that the user is already logged in)

**Reply by Kojirremer** [reply]
Are you sure they are registered and logged in? Depending on your settings, users can move around and do stuff even when unregistered, and even after doing /register you still need to /login afterwards.
SEconomy's usual way of doing things is to create an account on first login. I don't remember if you need any perms for that, however. May be worth checking.

**Reply by Ophise** [reply]
All users are registered and authorized 100%

**Reply by Ophise** [reply]
I checked personally by creating a second account

**Comment by Ophise** [comment]
I want to note that when authorizing, I write 2 messages:
1 that the user is authored. 
2 "There was an error proccesing your request"


My server is hosted. Server base - Linux
Maybe this will give more information.

**Reply by Kojirremer** [reply]
Sorry, I don't know if SEconomy works well on Linux or not. The only think I can suggest is to make sure that you have the latest version of Mono installed, check SEconomy.config.json to make sure BankAccountsEnabled is set to true, and to give the seconomy.world.mobgains permission to the group you want earning currency from killing mobs.

**Reply by theInterittus** [reply]
By any chance can you share where you found your version of SEconomy, been looking all night for a near recent version

**Reply by bartico6** [reply]
https://github.com/Quinci135/SEconomy is most likely to be a working version as its the newest of all forks I could find.

**Reply by theInterittus** [reply]
How exactly is it added to plugins, i tried adding all the dll's from Builds but it gives a ton of errors on server startup

**Reply by bartico6** [reply]
I can't tell you because I never used it. You will have to ask the author or read the build instructions, if there are any.

**Reply by Ophise** [reply]
The plugin can be said to work, but does not see the player database. sometimes he starts working and sees my account, once he was even able to see 6 accounts, but then he stopped seeing them again. I can't understand what it is connected with, and what kind of database Seconomy needs. I can't even find where the data with the money is stored ...

**Reply by Ophise** [reply]
maybe you know any other working and interesting plugins? Or is there any current plugin for the economy or trade? I watch other servers and they are taking them from somewhere.

**Reply by Ophise** [reply]
I have 4-5 plugins working for me ((

**Reply by Kojirremer** [reply]
Quinci's fork is indeed (as I already said twice before) the only public fork of SEconomy working right now, and since I've been using it for almost two months without issue, I'd say it works as intended. I also explained where to find the DLLs and where to put them. If you are getting errors then I'm sorry but I don't know what on your end is causing them, because I'm using the exact same build.

**Reply by Ophise** [reply]
Do you know if there is an actual plugin for the clan / guild system? I make the server as much RPG as possible, and I would like as many plugins as possible, could you give a list of plugins that you have installed? Or maybe you know the workers? It doesn't matter which ones, the main thing is that they work

**Reply by theInterittus** [reply]
> Quinci's fork is indeed (as I already said twice before) the only public fork of SEconomy working right now, and since I've been using it for almost two months without issue, I'd say it works as intended. I also explained where to find the DLLs and where to put them. If you are getting errors then I'm sorry but I don't know what on your end is causing them, because I'm using the exact same build.

I got the build working. Does SEconomy still have the ranking system, and if so how do I find the config for it

**Reply by Kojirremer** [reply]
Quoting my first message in this thread: "I'm sorry to say, though, that I don't know where you can find the script right now. It may be surprisingly easy to find, or someone may provide it for you. Mine has been tweaked for spanish language quite a bit, so it won't be of use to you.

EDIT: I forgot to put the name of the script: "ranking-system-2.0.js""

**Reply by theInterittus** [reply]
> Quoting my first message in this thread: "I'm sorry to say, though, that I don't know where you can find the script right now. It may be surprisingly easy to find, or someone may provide it for you. Mine has been tweaked for spanish language quite a bit, so it won't be of use to you.
> 
> EDIT: I forgot to put the name of the script: "ranking-system-2.0.js""

Two questions for you.
1. Did you put ranking-system-2.0.js into your serverscripts directory or SEconomy directory? Reason I ask this is because the Tshock thread and readme are entirely different.
2. What does your rank setup look like so I can reference it
`var rankingList = {
}
`

**Reply by Ophise** [reply]
> Quoting my first message in this thread: "I'm sorry to say, though, that I don't know where you can find the script right now. It may be surprisingly easy to find, or someone may provide it for you. Mine has been tweaked for spanish language quite a bit, so it won't be of use to you.
> 
> EDIT: I forgot to put the name of the script: "ranking-system-2.0.js""

As it turned out, I had a plugin with which Seconomy had a conflict, now everything works great! Thanks!

**Reply by Kojirremer** [reply]
> > Quoting my first message in this thread: "I'm sorry to say, though, that I don't know where you can find the script right now. It may be surprisingly easy to find, or someone may provide it for you. Mine has been tweaked for spanish language quite a bit, so it won't be of use to you.
> > EDIT: I forgot to put the name of the script: "ranking-system-2.0.js""
> 
> Two questions for you.
> 
> 1. Did you put ranking-system-2.0.js into your serverscripts directory or SEconomy directory? Reason I ask this is because the Tshock thread and readme are entirely different.
> 2. What does your rank setup look like so I can reference it
>    `var rankingList = { } `

All JIST scripts go to the serverscripts directory, including ranking-system-2.0.js. As for the second question, I don't know what you want me to do. It's hundreds of lines long even without a single rank configured. I think knowing the name of the script should be enough to search for it.

**Reply by theInterittus** [reply]
> > > Quoting my first message in this thread: "I'm sorry to say, though, that I don't know where you can find the script right now. It may be surprisingly easy to find, or someone may provide it for you. Mine has been tweaked for spanish language quite a bit, so it won't be of use to you.
> > > EDIT: I forgot to put the name of the script: "ranking-system-2.0.js""
> > 
> > 
> > Two questions for you.
> > 
> > 1. Did you put ranking-system-2.0.js into your serverscripts directory or SEconomy directory? Reason I ask this is because the Tshock thread and readme are entirely different.
> > 2. What does your rank setup look like so I can reference it
> >    `var rankingList = { } `
> 
> All JIST scripts go to the serverscripts directory, including ranking-system-2.0.js. As for the second question, I don't know what you want me to do. It's hundreds of lines long even without a single rank configured. I think knowing the name of the script should be enough to search for it.

Sorry I'll rephrase that, I wonder how the format looks for it.
Theres a link that references how to set up the ranking list, but it doesn't exist anymore.
```
/*
 * Please see the forums at http://plugins.tw.id.au/ for tutorials on how 
 * to make a ranking list. 
 */
```

**Reply by Kojirremer** [reply]
https://web.archive.org/web/20150714210248/http://plugins.tw.id.au/files/scripts/ranking-system-2.0.js
That's what I found after a quick search. That's as far as I'm willing to go.

Edit: I was sure there was at least one example of a level/rank. I'll try checking mine,but I'm at work, so it may or may not take some time.

**Reply by Kojirremer** [reply]
var rankingList = {
    "Nivel1": {
        "name": "Nivel1",
        "parentgroup": undefined,
        "levelupcommands": [
			"/i 2334 1",
            "/i 3501 1",
            "/i 3503 1",
            "/i 3500 1",
            "/i 1774 5",
			"/i 3544 5",
            "/i 287 100"
        ],
        "group": "nivel1",
        "cost": "0"
    },
    "Nivel2": {
        "name": "Nivel2",
        "parentgroup": "nivel1",
        "levelupcommands": [
			"/i 2334 1",
			"/i 3544 5",
            "/i 287 100"
        ],
        "group": "nivel2",
        "cost": "190"
    },

Hope that's enough, 'cause I'm really bailing for real now if it isn't.

**Reply by theInterittus** [reply]
> var rankingList = {
> "Nivel1": {
> "name": "Nivel1",
> "parentgroup": undefined,
> "levelupcommands": [
> "/i 2334 1",
> "/i 3501 1",
> "/i 3503 1",
> "/i 3500 1",
> "/i 1774 5",
> "/i 3544 5",
> "/i 287 100"
> ],
> "group": "nivel1",
> "cost": "0"
> },
> "Nivel2": {
> "name": "Nivel2",
> "parentgroup": "nivel1",
> "levelupcommands": [
> "/i 2334 1",
> "/i 3544 5",
> "/i 287 100"
> ],
> "group": "nivel2",
> "cost": "190"
> },
> 
> Hope that's enough, 'cause I'm really bailing for real now if it isn't.

Thanks for the reply, that information is good enough

**Reply by Ophise** [reply]
I installed the Seconomy plugin, but I don't understand in any way where the information about the players' currency is transferred and stored.

**Reply by Ophise** [reply]
> var rankingList = {
> "Nivel1": {
> "name": "Nivel1",
> "parentgroup": undefined,
> "levelupcommands": [
> "/i 2334 1",
> "/i 3501 1",
> "/i 3503 1",
> "/i 3500 1",
> "/i 1774 5",
> "/i 3544 5",
> "/i 287 100"
> ],
> "group": "nivel1",
> "cost": "0"
> },
> "Nivel2": {
> "name": "Nivel2",
> "parentgroup": "nivel1",
> "levelupcommands": [
> "/i 2334 1",
> "/i 3544 5",
> "/i 287 100"
> ],
> "group": "nivel2",
> "cost": "190"
> },
> 
> Hope that's enough, 'cause I'm really bailing for real now if it isn't.

I installed the Seconomy plugin, but I don't understand in any way where the information about the players' currency is transferred and stored.
